### PR TITLE
Macos universal build

### LINKED ISSUE #98: Support universal macOS build
Currently the `libGodotFmod.osx.release.64.dylib` library for macOS is only built for the x86_64 architecture and not the arm64 (apple silicon) architecture. Having a universal build will support both of these architectures.

Both FMOD and godot-cpp (3.4-stable) are built as universal and support both architectures.

The current CI process has godot-cpp tagged at 3.3.4-stable, which unfortunately does not include the universal build changes. So the CI process will need to be moved forward to 3.4-stable to support this work. I attempted to bring in the macOS universal support with #97, which should still be valid when the repo is aligned with godot-cpp version 3.4. Those changes worked for me locally because I had godot-cpp checked out to master and didn't realize the CI process was tagged at version 3.3.4.

The latest release in [utopia-rise/godot-cpp](https://github.com/utopia-rise/godot-cpp) is 3.3.4-stable so the 3.4-stable version is not yet available.

@piiertho or @CedNaru do you know when a 3.4-stable version of the utopia-rise/godot-cpp fork will be available? I'm happy to help with this, but not sure where to start with that piece.

Thanks for the great work on this project!

### PR DESCRIPTION
**Pull request by langfort**
Closes #98

### CODE REVIEW

**Review by CedNaru — APPROVED** [review]

**Comment by CedNaru** [comment]
LGTM. Wait for piertho's approval so he can test on his side.

**Review by piiertho — APPROVED** [review]
LGTM

**Pull request merged**
